src/main/scala/co/teapot/mmalloc/MemoryMappedAllocator.scala

```diff
@@ -22,20 +22,86 @@ import it.unimi.dsi.fastutil.longs.LongArrayList
 
 import MemoryMappedAllocator._ // Make constants available
 
+/**
+  * A class wrapping a raw Long that counts bytes (in an array,
+  * memory block, per-node, etc.). This class exists mostly to
+  * support Offset and Pointer manipulation with explicit
+  * emphasis that things you manipulate are expressed in bytes.
+  */
+case class ByteCount(raw: Long) extends Ordered[ByteCount] {
+  override def compare(that: ByteCount): Int = raw.compareTo(that.raw)
+  def toOffset: Offset = Offset(raw)
+  def +(other: ByteCount): ByteCount = ByteCount(raw + other.raw)
+}
+object ByteCount {
+  implicit val ordering: Ordering[ByteCount] = Ordering.by(_.raw)
+  def forNodes(nodesNumber: Int): ByteCount = ByteCount(4L * nodesNumber)
+}
+
+/**
+  * A class wrapping a raw Long that keeps track of an offset
+  * relative to some pointer. It defines some handy utilities
+  * for basic algebra on offsets and pointers.
+  *
+  * To reduce risk of integer overflow, it doesn't define
+  * multiplication methods. Use factory methods in companion object
+  * instead.
+  */
+case class Offset(raw: Long) {
+  require(raw >= 0)
+  def +(other: Offset): Offset = Offset(raw + other.raw)
+  def +(other: Pointer): Pointer = Pointer(raw + other.raw)
+  def -(other: Offset): Offset = Offset(raw - other.raw)
+  def toPointer: Pointer = Pointer(raw)
+  def toBlockNumber(blockSize: ByteCount): Int = {
+    val result = raw / blockSize.raw
+    assert(result < Int.MaxValue)
+    result.toInt
+  }
+  def isStartOfBlock(blockSize: ByteCount): Boolean = {
+    raw % blockSize.raw == 0
+  }
+  def toByteCount: ByteCount = ByteCount(raw)
+}
+object Offset {
+  def fromBlocks(blockCount: Int, blockSize: ByteCount): Offset =
```

Comment:
In line with `ints` and `longs`, would `blocks` be a more consistent method name here?